conda_lock/src_parser/__init__.py

```diff
@@ -101,6 +120,89 @@ def validate_hash(cls, v: HashModel, values: Dict[str, typing.Any]) -> HashModel
         return v
 
 
+class TimeMeta(StrictModel):
+    """Stores information about when the lockfile was generated."""
+
+    created_at: str = Field(..., description="Time stamp of lock-file creation time")
+
+    @classmethod
+    def create(cls) -> "TimeMeta":
+        import time
+
+        return cls(
+            created_at=datetime.datetime.utcnow().isoformat(timespec="seconds") + "Z"
+        )
+
+
+class GitMeta(StrictModel):
+    """
+    Stores information about the git repo the lockfile is being generated in (if applicable) and
+    the git user generating the file.
+    """
+
+    git_user_name: Optional[str] = Field(
+        default=None, description="Git user.name field of global config"
+    )
+    git_user_email: Optional[str] = Field(
+        default=None, description="Git user.email field of global config"
+    )
+    git_sha: Optional[str] = Field(
+        default=None, description="sha256 hash of the most recent git commit"
+    )
+
+    @classmethod
+    def create(cls) -> "GitMeta":
```

Comment:
Do we want to have the git metadata associated with the HEAD commit for the relevant repo or should it be associated with the most recent commit for the input files.  

The latter would mean that if the components of a lock don't change then the git_meta is unaffected